tests/unittests/classification/test_group_fairness.py

```diff
@@ -0,0 +1,136 @@
+from scipy.special import expit as sigmoid
+
+from torchmetrics.classification.group_fairness import BinaryFairness
+from torchmetrics.functional.classification.group_fairness import binary_fairness
+from unittests.classification.inputs import _group_cases
+from unittests.helpers import seed_all
+from unittests.helpers.testers import THRESHOLD, MetricTester, inject_ignore_index, remove_ignore_index_groups
+
+seed_all(42)
+
+
+def _fairlearn_binary(preds, target, groups, ignore_index):
+    metrics = {"dp": selection_rate, "eo": true_positive_rate}
+
+    preds = preds.numpy()
+    target = target.numpy()
+    groups = groups.numpy()
+
+    if np.issubdtype(preds.dtype, np.floating):
+        if not ((preds > 0) & (preds < 1)).all():
+            preds = sigmoid(preds)
+        preds = (preds >= THRESHOLD).astype(np.uint8)
+
+    target, preds, groups = remove_ignore_index_groups(target, preds, groups, ignore_index)
+
+    mf = MetricFrame(metrics=metrics, y_true=target, y_pred=preds, sensitive_features=groups)
+
+    mf_group = mf.by_group
+    ratios = mf.ratio()
+
+    return {
+        f"DP_{mf_group['dp'].idxmin()}_{mf_group['dp'].idxmax()}": torch.tensor(ratios["dp"], dtype=torch.float),
+        f"EO_{mf_group['eo'].idxmin()}_{mf_group['eo'].idxmax()}": torch.tensor(ratios["eo"], dtype=torch.float),
+    }
+
+
+@pytest.mark.parametrize("input", _group_cases)
+class TestBinaryFairness(MetricTester):
+    @pytest.mark.parametrize("ignore_index", [None, 0, -1])
+    @pytest.mark.parametrize("ddp", [False, True])
+    def test_binary_fairness(self, ddp, input, ignore_index):
+        preds, target, groups = input
+        if ignore_index == -1:
+            target = inject_ignore_index(target, ignore_index)
+
+        self.run_class_metric_test(
+            ddp=ddp,
+            preds=preds,
+            target=target,
+            metric_class=BinaryFairness,
+            reference_metric=partial(_fairlearn_binary, ignore_index=ignore_index),
+            metric_args={"threshold": THRESHOLD, "ignore_index": ignore_index, "num_groups": 2, "task": "all"},
+            groups=groups,
+            fragment_kwargs=True,
+        )
+
+    @pytest.mark.parametrize("ignore_index", [None, 0, -1])
+    def test_binary_fairness_functional(self, input, ignore_index):
+        preds, target, groups = input
+        if ignore_index == -1:
+            target = inject_ignore_index(target, ignore_index)
+
+        self.run_functional_metric_test(
+            preds=preds,
+            target=target,
+            metric_functional=binary_fairness,
+            reference_metric=partial(_fairlearn_binary, ignore_index=ignore_index),
+            metric_args={
+                "threshold": THRESHOLD,
+                "ignore_index": ignore_index,
+                "task": "all",
+            },
+            groups=groups,
+            fragment_kwargs=True,
+        )
+
+    # def test_binary_fairness_differentiability(self, input):
```

Comment:
Why this is commented, then remove it?

Reply:
Sorry, was temporary since they were still failing. Fixed it now.